src/server/scripts/Spells/spell_rogue.cpp

```diff
@@ -542,43 +542,66 @@ class spell_rog_prey_on_the_weak : public AuraScript
 };
 
 // -1943 - Rupture
-class spell_rog_rupture : public AuraScript
+#define RuptureScriptName "spell_rog_rupture"
+class spell_rog_rupture : public SpellScriptLoader
 {
-    PrepareAuraScript(spell_rog_rupture);
+    public:
+        spell_rog_rupture() : SpellScriptLoader(RuptureScriptName) { }
 
-    bool Load() override
+    class spell_rog_rupture_AuraScript : public AuraScript
     {
-        Unit* caster = GetCaster();
-        return caster && caster->GetTypeId() == TYPEID_PLAYER;
-    }
+        public:
+            // For Glyph of Backstab use
+            uint32 BonusDuration;
 
-    void CalculateAmount(AuraEffect const* /*aurEff*/, int32& amount, bool& canBeRecalculated)
-    {
-        if (Unit* caster = GetCaster())
+        PrepareAuraScript(spell_rog_rupture_AuraScript);
+
+        bool Load() override
         {
-            canBeRecalculated = false;
+            Unit* caster = GetCaster();
+            BonusDuration = 0;
+            return caster && caster->GetTypeId() == TYPEID_PLAYER;
+        }
+
+        void ResetDuration(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+        {
+            BonusDuration = 0;
+        }
 
-            float const attackpowerPerCombo[6] =
+        void CalculateAmount(AuraEffect const* /*aurEff*/, int32& amount, bool& canBeRecalculated)
+        {
+            if (Unit* caster = GetCaster())
             {
-                0.0f,
-                0.015f,         // 1 point:  ${($m1 + $b1*1 + 0.015 * $AP) * 4} damage over 8 secs
-                0.024f,         // 2 points: ${($m1 + $b1*2 + 0.024 * $AP) * 5} damage over 10 secs
-                0.03f,          // 3 points: ${($m1 + $b1*3 + 0.03 * $AP) * 6} damage over 12 secs
-                0.03428571f,    // 4 points: ${($m1 + $b1*4 + 0.03428571 * $AP) * 7} damage over 14 secs
-                0.0375f         // 5 points: ${($m1 + $b1*5 + 0.0375 * $AP) * 8} damage over 16 secs
-            };
-
-            uint8 cp = caster->ToPlayer()->GetComboPoints();
-            if (cp > 5)
-                cp = 5;
-
-            amount += int32(caster->GetTotalAttackPowerValue(BASE_ATTACK) * attackpowerPerCombo[cp]);
+                canBeRecalculated = false;
+
+                float const attackpowerPerCombo[6] =
+                        {
+                                0.0f,
+                                0.015f,         // 1 point:  ${($m1 + $b1*1 + 0.015 * $AP) * 4} damage over 8 secs
+                                0.024f,         // 2 points: ${($m1 + $b1*2 + 0.024 * $AP) * 5} damage over 10 secs
+                                0.03f,          // 3 points: ${($m1 + $b1*3 + 0.03 * $AP) * 6} damage over 12 secs
+                                0.03428571f,    // 4 points: ${($m1 + $b1*4 + 0.03428571 * $AP) * 7} damage over 14 secs
+                                0.0375f         // 5 points: ${($m1 + $b1*5 + 0.0375 * $AP) * 8} damage over 16 secs
+                        };
```

Comment:
Each call to CalculateAmount creates an attackpowerPerCombo array. It's always the same. I would make it static